packages/widget/src/controllers/transfers/evm/execute.ts

```diff
@@ -28,14 +32,31 @@ export async function executeNextEvmTransaction(
       this.host.requestUpdate();
       await tx.wait();
       this.pendingEvmApprovalTransactions.shift();
+
+      transactions.push(
+        ...this.pendingEvmApprovalTransactions,
+        this.pendingTransferTransaction
+      );
+
+      this.estimatedGas = await estimateEvmGas(
+        this.sourceNetwork?.chainId as number,
+        this.walletContext.value?.evmWallet?.provider as Eip1193Provider,
+        this.walletContext.value?.evmWallet?.address as string,
```

Comment:
this is more or less the same as having non-null assertions when optional chaining. doesn't seem very safe (at least without value checking first, which would then make these assertions redundant). are we 100% sure they are non-nullish?

Reply:
These values are already checked before

Reply:
i see, in the future maybe it'd be better to do a check using type guard (to narrow down type to a one with non-nullish values), because this causes a lot of cognitive load/unnecessary checks/type assertions